compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -3255,19 +3255,24 @@ TR::Register *OMR::X86::TreeEvaluator::arraytranslateEvaluator(TR::Node *node, T
    TR::Register *srcPtrReg, *dstPtrReg, *transTableReg, *termCharReg, *lengthReg;
    bool stopUsingCopyReg1 = TR::TreeEvaluator::stopUsingCopyRegAddr(node->getChild(0), srcPtrReg, cg);
    bool stopUsingCopyReg2 = TR::TreeEvaluator::stopUsingCopyRegAddr(node->getChild(1), dstPtrReg, cg);
-   bool stopUsingCopyReg4 = TR::TreeEvaluator::stopUsingCopyRegInteger(node->getChild(3), termCharReg, cg);
+   bool stopUsingCopyReg4 = false;
+   if (!sourceByte)
+      {
+      stopUsingCopyReg4 = TR::TreeEvaluator::stopUsingCopyRegInteger(node->getChild(3), termCharReg, cg);
+      }
```

Comment:
"dummy" means that the value is not used in the helper (It is a free register). In the old helper, when sourceBye is false,  we overwrite child(3) register in the very beginning  and we just use it as a free register. But in the new helper, I realized that we have other free registers to use so we don't actually need to evaluate child(3).